integrations/google_ai/src/haystack_integrations/components/embedders/google_ai/text_embedder.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import warnings
+from typing import Any, Dict, List, Literal, Optional
+
+from google import genai
+from google.genai import types
+from haystack.core.component import component
+from haystack.core.serialization import default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+# Load environment variables from .env file, if present
+logger = logging.getLogger(__name__)
+
+
+def resolve_secret(secret: Optional[Secret]) -> Optional[str]:
+    return secret.resolve_value() if secret else None
+
+
+@component
+class GoogleAIGeminiTextEmbedder:
+    """
+    A component for embedding text using Google AI models (e.g., Gemini).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.embedders.google_ai import GoogleAIGeminiTextEmbedder
+    from haystack.utils.auth import Secret
+
+    # Ensure GOOGLE_API_KEY environment variable is set
+
+    embedder = GoogleAIGeminiTextEmbedder(model="gemini-embedding-exp-03-07", task_type="retrieval_document")
+    embedder.warm_up()
+
+    text = "What is the meaning of life?"
+    result = embedder.run([text])
+    print(result['embedding'])
+    print(result['meta'])
+
+    # Example with explicit API key
+    embedder_explicit_key = GoogleAIGeminiTextEmbedder(
+        api_key=Secret.from_token("YOUR_GOOGLE_API_KEY"),
+        model="models/embedding-001",
+        task_type="retrieval_query"
+    )
+    embedder_explicit_key.warm_up()
+    result_query = embedder_explicit_key.run(["How does quantum physics work?"])
+
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Literal[
+            "gemini-embedding-exp-03-07", "text-embedding-004", "embedding-001"
+        ] = "gemini-embedding-exp-03-07",
+        api_key: Optional[Secret] = Secret.from_env_var("GEMINI_API_KEY"),  # noqa: B008
+        task_type: Optional[str] = "retrieval_document",
+        # Supported task types: "retrieval_query", "retrieval_document", "semantic_similarity",
+        #                       "classification", "clustering", "question_answering", "fact_verification"
+        # See: https://ai.google.dev/docs/embeddings#task_types
+        title: Optional[str] = None,  # Relevant only for task_type="retrieval_document"
+        output_dimensionality: Optional[int] = None,
+    ):
+        """
+        Initializes the GoogleAIGeminiTextEmbedder component.
+
+        :param model: The name of the Google AI embedding model to use.
+                      Defaults to "models/embedding-001".
+        :param api_key: The Google AI API key. It can be explicitly provided or automatically read from the
+                        `GOOGLE_API_KEY` environment variable.
```

Comment:
Note: above you're initializing this from `GEMINI_API_KEY`, but here you are referring to `GOOGLE_API_KEY`.